addons/product_margin/product_margin.py

```diff
@@ -71,68 +48,44 @@ def _product_margin(self, cr, uid, ids, field_names, arg, context=None):
-            help="Sum of Multiplication of Sale Catalog price and quantity of Customer Invoices"),
-        'normal_cost'  : fields.function(_product_margin, type='float', string='Normal Cost', multi='product_margin',
-            help="Sum of Multiplication of Cost price and quantity of Supplier Invoices"),
-        'total_margin' : fields.function(_product_margin, type='float', string='Total Margin', multi='product_margin',
-            help="Turnover - Standard price"),
-        'expected_margin' : fields.function(_product_margin, type='float', string='Expected Margin', multi='product_margin',
-            help="Expected Sale - Normal Cost"),
-        'total_margin_rate' : fields.function(_product_margin, type='float', string='Total Margin Rate(%)', multi='product_margin',
-            help="Total margin * 100 / Turnover"),
-        'expected_margin_rate' : fields.function(_product_margin, type='float', string='Expected Margin (%)', multi='product_margin',
-            help="Expected margin * 100 / Expected Sale"),
-    }
+    date_from = fields.Date(compute='_product_margin', string='Margin Date From')
+    date_to = fields.Date(compute='_product_margin', string='Margin Date To')
+    invoice_state = fields.Selection(compute='_product_margin', selection=[('paid', 'Paid'), ('open_paid', 'Open and Paid'),
+                                    ('draft_open_paid', 'Draft, Open and Paid')], string='Invoice State', readonly=True)
+    sale_avg_price = fields.Float(compute='_product_margin', string='Avg. Unit Price', help="Avg. Price in Customer Invoices.")
+    purchase_avg_price = fields.Float(compute='_product_margin', string='Avg. Unit Price', help="Avg. Price in Supplier Invoices ")
+    sale_num_invoiced = fields.Float(compute='_product_margin', string='# Invoiced in Sale', help="Sum of Quantity in Customer Invoices")
+    purchase_num_invoiced = fields.Float(compute='_product_margin', string='# Invoiced in Purchase', multi='product_margin', help="Sum of Quantity in Supplier Invoices")
+    sales_gap = fields.Float(compute='_product_margin', string='Sales Gap', help="Expected Sale - Turn Over")
+    purchase_gap = fields.Float(compute='_product_margin', string='Purchase Gap', help="Normal Cost - Total Cost")
+    turnover = fields.Float(compute='_product_margin', string='Turnover', help="Sum of Multiplication of Invoice price and quantity of Customer Invoices")
+    total_cost = fields.Float(compute='_product_margin', string='Total Cost', help="Sum of Multiplication of Invoice price and quantity of Supplier Invoices ")
+    sale_expected = fields.Float(compute='_product_margin', string='Expected Sale', help="Sum of Multiplication of Sale Catalog price and quantity of Customer Invoices")
+    normal_cost = fields.Float(compute='_product_margin', string='Normal Cost', help="Sum of Multiplication of Cost price and quantity of Supplier Invoices")
```

Comment:
same for normal cost no need to use string.. ,same for total cost, turnover, Purchase gap, sales gap,